src/mountscript.sh

```diff
@@ -610,59 +983,127 @@ if ! ensure_aznfswatchdog; then
 if [ $? -ne 0 ]; then
-    echo "$nfs_host"
+    echo "$nfs_vers"
     exit 1
 fi
 
-if ! is_valid_blob_fqdn "$nfs_host"; then
-    eecho "Not a valid Azure Blob NFS endpoint: ${nfs_host}!"
-    eecho "Must be of the form 'account.blob.core.windows.net'!"
-    exit 1
+if [ $nfs_vers == "4.1" ]; then
+    AZ_PREFIX="file"
+else
+    AZ_PREFIX="blob"
 fi
 
-# Resolve the IP address for the NFS host
-nfs_ip=$(resolve_ipv4 "$nfs_host" "true")
+nfs_host=$(get_host_from_share "$1" "$AZ_PREFIX")
 if [ $? -ne 0 ]; then
-    echo "$nfs_ip"
-    eecho "Cannot resolve IP address for ${nfs_host}!"
-    eecho "Mount failed!"
+    echo "$nfs_host"
+    exit 1
+fi
+
+# TODO: Comment out below code for devfabric. 'is_valid_fqdn' will fail on devfabric.
+if ! is_valid_fqdn "$nfs_host" "$AZ_PREFIX"; then
+    eecho "Not a valid Azure $AZ_PREFIX NFS endpoint: ${nfs_host}!"
+    eecho "Must be of the form 'account.$AZ_PREFIX.core.windows.net'!"
     exit 1
 fi
 
-nfs_dir=$(get_dir_from_share "$1")
+nfs_dir=$(get_dir_from_share "$1" "$AZ_PREFIX")
 if [ $? -ne 0 ]; then
     echo "$nfs_dir"
     exit 1
 fi
 
 if [ -z "$nfs_dir" ]; then
     eecho "Bad share name: ${1}!"
-    eecho "Share to be mounted must be of the form 'account.blob.core.windows.net:/account/container'!"
+    eecho "Share to be mounted must be of the form 'account.$AZ_PREFIX.core.windows.net:/account/container'!"
     exit 1
 fi
 
-mount_point="$2"
+if [ "$nfs_vers" == "4.1" ]; then
+    vecho "nfs_host=[$nfs_host], nfs_dir=[$nfs_dir], mount_point=[$mount_point], options=[$OPTIONS], mount_options=[$MOUNT_OPTIONS]."
 
-OPTIONS=
-MOUNT_OPTIONS=
+    # AZ_FILES_MOUNTMAP file must have been created by aznfswatchdog service.
+    if [ ! -f $AZ_FILES_MOUNTMAP ]; then
+        touch $AZ_FILES_MOUNTMAP
+        if [ $? -ne 0 ]; then
+            eecho "[FATAL] Not able to create '${AZ_FILES_MOUNTMAP}'!"
+            exit 1
+        fi
+    fi
 
-parse_arguments $*
+    if ! chattr -f +i $AZ_FILES_MOUNTMAP; then
+        wecho "chattr does not work for ${AZ_FILES_MOUNTMAP}!"
+    fi
+
+    if [[ "$MOUNT_OPTIONS" == *"notls"* ]]; then
+        vecho "notls option is enabled. Mount nfs share without TLS."
+
+        if [[ "$MOUNT_OPTIONS" == *"notls,"* ]]; then
+            MOUNT_OPTIONS=${MOUNT_OPTIONS//notls,/}
+        else
+            MOUNT_OPTIONS=${MOUNT_OPTIONS//,notls/}
+        fi
+
+        # Do the actual mount.
+        mount_output=$(mount -t nfs -o "$MOUNT_OPTIONS" "${nfs_host}:${nfs_dir}" "$mount_point" 2>&1)
+        mount_status=$?
+
+        if [ -n "$mount_output" ]; then
+            pecho "$mount_output"
+            vecho "Mount completed: ${nfs_host}:${nfs_dir} on $mount_point"
+        fi
+
+        if [ $mount_status -ne 0 ]; then
+            eecho "Mount failed!"
+            exit 1
+        fi
+    else
+        vecho "Mount nfs share with TLS."
+
+        IFS=/ read _ account container extra <<< "$nfs_dir"
+        storageaccount="$account"
+
+        tls_nfsv4_files_share_mount
```

Comment:
should you check the error code here ?

Reply:
not required. We anyway exit after calling this function for v4. Also, in this function, if anything fails, we exit from there so there is no need to check the error code here.